spec/unit/conjur/puppet_module/http_spec.rb

```diff
@@ -138,20 +155,15 @@ def http_unauthorized
     end
 
     it 'handles retrieval errors' do
-      expect(mock_connection).to receive(:post).with('/' + target_path, mock_post_data)
+      expect(mock_connection).to receive(:post).with('/' + target_path, mock_post_data, {})
                                                .and_return(http_unauthorized)
 
       expect { subject.post(target_url, target_path, ssl_certificate, mock_post_data) }
         .to raise_error Net::HTTPError
     end
 
```

Comment:
It's not immediately clear to me if/how the removed test is no longer relevant ?

Reply:
Old test that was removed would raise error when no form data in the http content was presented. Since it's possible to pass in the data over the endpoint URI (e.g. `foo.com/my/path?something=bar` which would fail it, this logic path was not really valid and the test needed to be removed. The added test is to verify that the header hash is passed and is not directly related to the aforementioned removal of a test.